numba/np/arrayobj.py

```diff
@@ -1389,6 +1409,14 @@ def impl(array, shape):
     elif isinstance(shape, types.Integer):
         def impl(array, shape):
             return np.broadcast_to(array, (shape,))
+    elif isinstance(shape, types.Tuple):
```

Comment:
This branch could use a simple check to see if it's actually an empty tuple and not something like `(1, 'x')`, the current logic accepts that shape which is invalid.

Reply:
Thanks, fixed!